Userland/Libraries/LibWeb/HTML/Window.cpp

```diff
@@ -1437,14 +1442,124 @@ OrderedHashMap<String, JS::NonnullGCPtr<Navigable>> Window::document_tree_child_
+        JS::GCPtr<NavigableContainer> container = nullptr;
+        const_cast<Window&>(*this).associated_document().for_each_in_subtree_of_type<HTML::NavigableContainer>([&](HTML::NavigableContainer& navigable_container) {
+            if (!navigable_container.content_navigable())
+                return IterationDecision::Continue;
+            if (objects.navigables.contains_slow(JS::NonnullGCPtr { *navigable_container.content_navigable() })) {
+                container = navigable_container;
+                return IterationDecision::Break;
+            }
+            return IterationDecision::Continue;
+        });
+        // 2. Return container's content navigable's active WindowProxy.
+        VERIFY(container);
+        return container->content_navigable()->active_window_proxy();
+    }
+
+    // 3. Otherwise, if objects has only one element, return that element.
+    if (objects.elements.size() == 1)
+        return objects.elements[0];
+
+    // 4. Otherwise return an HTMLCollection rooted at window's associated Document,
+    //    whose filter matches only named objects of window with the name name. (By definition, these will all be elements.)
+    return DOM::HTMLCollection::create(const_cast<Window&>(*this).associated_document(), DOM::HTMLCollection::Scope::Descendants, [name](auto& element) -> bool {
+        if ((is<HTMLEmbedElement>(element) || is<HTMLFormElement>(element) || is<HTMLImageElement>(element) || is<HTMLObjectElement>(element))
+            && (element.attribute(AttributeNames::name) == name.view()))
+            return true;
+        return element.attribute(AttributeNames::id) == name.view();
+    });
+}
+
+// https://html.spec.whatwg.org/#dom-window-nameditem-filter
+Window::NamedObjects Window::named_objects(StringView name)
+{
+    // NOTE: Since the Window interface has the [Global] extended attribute, its named properties
+    //       follow the rules for named properties objects rather than legacy platform objects.
+
+    // Named objects of Window object window with the name name, for the purposes of the above algorithm, consist of the following:
+    NamedObjects objects;
+
+    // document-tree child navigables of window's associated Document whose target name is name;
+    auto children = associated_document().document_tree_child_navigables();
+    for (auto& navigable : children) {
+        if (navigable->target_name() == name) {
+            objects.navigables.append(*navigable);
+        }
+    }
+
+    // embed, form, img, or object elements that have a name content attribute whose value is name
+    // and are in a document tree with window's associated Document as their root; and
+    // HTML elements that have an id content attribute whose value is name and are in a document tree with window's associated Document as their root.
+    associated_document().for_each_in_subtree([&objects, &name, this](auto& node) -> IterationDecision {
+        if (is<HTMLElement>(node)) {
```

Comment:
You can use `for_each_in_subtree_of_type<HTMLElement>(...)`

Although.. are we supposed to exclude `SVGElement`s here? Otherwise maybe `for_each_in_subtree_of_type<Element>` is more appropriate.

Reply:
hm. looks like we are supposed to include SVGElement? 
![image](https://github.com/SerenityOS/serenity/assets/8388494/c97d0eae-f690-467c-b0ff-ead38ec6c698)


Though the spec says...

> the value of the [id](https://html.spec.whatwg.org/#the-id-attribute) content attribute for all [HTML elements](https://html.spec.whatwg.org/#html-elements) that have a non-empty [id](https://html.spec.whatwg.org/#the-id-attribute) content attribute and are [in a document tree](https://dom.spec.whatwg.org/#in-a-document-tree) with window's [associated Document](https://html.spec.whatwg.org/#concept-document-window) as their [root](https://dom.spec.whatwg.org/#concept-tree-root).

> To ease migration from HTML to XML, user agents conforming to this specification will place elements in HTML in the [http://www.w3.org/1999/xhtml](https://infra.spec.whatwg.org/#html-namespace) namespace, at least for the purposes of the DOM and CSS. The term "HTML elements" refers to any element in that namespace, even in XML documents.

So... any element in the xhtml namespace? Idk if SVG elements are in that namespace